### DISCUSSION TITLE
Change working directory to script directory

### DISCUSSION THREAD

**Discussion by herrwinfried** [LSPosed/MagiskOnWSALocal#148] [Ideas]
### Steps to reproduce/复现步骤

1.  run run.sh
2. select the required options.
3. watch the screen. WSA throws an error when it comes to unzip.

### Expected behaviour/预期行为

It should complete the process and unzip the zip.

### Actual behaviour/实际行为

does not install. It does not allocate files.

### MagiskOnWSALocal commit full SHA/MagiskOnWSALocal 提交的完整哈希

f59f0120068620e2668134542aa73668a15256a5

### Linux distribution info/Linux 发行版信息

opensuse tumbleweed WSL

### Windows version/Windows 版本

10.0.22621.607

### Build Parameters/构建参数

COMMAND_LINE=--arch x64 --release-type retail --magisk-ver stable --gapps-brand OpenGApps --gapps-variant pico  --root-sol magisk --compress
INFO: Architecture: x64
INFO: Release Type: retail
INFO: Magisk Version: stable
INFO: GApps Brand: OpenGApps
INFO: GApps Variant: pico
INFO: Root Solution: magisk

### Version requirement/版本要求

- [X] I am using latest commit of MagiskOnWSALocal/我正在使用最新 commit 的版本

### Logs/日志

<details>
<summary>Details</summary>

(OK):download completed.
Extract WSA
unzipping to /tmp/wsa-build-ONXVoimmxy_/wsa
unzipping from /tmp/wsa-build-ONXVoimmxy_/wsa/WsaPackage_2207.40000.8.0_x64_Release-Nightly.msix
Traceback (most recent call last):
  File "/tmp/MagiskOnWSALocal/scripts/extractWSA.py", line 72, in <module>
    zip.extractall(workdir / arch)
  File "/usr/lib64/python3.10/zipfile.py", line 1645, in extractall
    self._extract_member(zipinfo, path, pwd)
  File "/usr/lib64/python3.10/zipfile.py", line 1700, in _extract_member
    shutil.copyfileobj(source, target)
  File "/usr/lib64/python3.10/shutil.py", line 198, in copyfileobj
    fdst_write(buf)
OSError: [Errno 28] No space left on device
Unzip WSA failed, is the download incomplete?
Build: an error has occurred, exit
Cleanup Download Directory
</details>

**Comment by Howard20181** [comment]
`OSError: [Errno 28] No space left on device`

Checking the amount of disk space remaining
`df -h`

**Comment by herrwinfried** [comment]
> `OSError: [Errno 28] No space left on device`
> 
> Checking the amount of disk space remaining `df -h`
```bash
❯ df -h
Filesystem      Size  Used Avail Use% Mounted on
none            3.9G  4.0K  3.9G   1% /mnt/wsl
drivers         477G  273G  205G  58% /usr/lib/wsl/drivers
none            3.9G     0  3.9G   0% /usr/lib/wsl/lib
/dev/sdc       1007G  9.8G  946G   2% /
none            3.9G  132K  3.9G   1% /mnt/wslg
rootfs          3.9G  1.9M  3.9G   1% /init
none            3.9G     0  3.9G   0% /dev
none            3.9G  620K  3.9G   1% /run
none            3.9G     0  3.9G   0% /run/lock
none            3.9G     0  3.9G   0% /run/shm
none            3.9G     0  3.9G   0% /run/user
tmpfs           4.0M     0  4.0M   0% /sys/fs/cgroup
none            3.9G   76K  3.9G   1% /mnt/wslg/versions.txt
none            3.9G   76K  3.9G   1% /mnt/wslg/doc
tmpfs           3.9G  2.9M  3.9G   1% /var/tmp
tmpfs           3.9G  358M  3.5G  10% /tmp
drvfs           477G  273G  205G  58% /mnt/c
```
and windows says 204 GB more free space.

**Comment by Howard20181** [comment]
The script uses `/tmp` as a temporary directory by default, if there is not enough space in that directory you can go to https://github.com/LSPosed/MagiskOnWSALocal/blob/f59f0120068620e2668134542aa73668a15256a5/scripts/build.sh#L32 to remove the `#` in front of that line to use the directory where the script is located as a temporary directory.

**Reply by herrwinfried** [reply]
> The script uses `/tmp` as a temporary directory by default, if there is not enough space in that directory you can go to
> 
> https://github.com/LSPosed/MagiskOnWSALocal/blob/f59f0120068620e2668134542aa73668a15256a5/scripts/build.sh#L32
> 
> to remove the `#` in front of that line to use the directory where the script is located as a temporary directory.

![image](https://user-images.githubusercontent.com/52379312/192294939-80a3260f-7ad6-4b7c-b0b3-c8055512f568.png)

How can I fix? wsa is not installed.

**Reply by Howard20181** [reply]
Win+R run `eventvwr.msc` to open the event viewer
In the left column find Application and Service Logs -> Microsoft -> Windows -> AppXDeployment-Server -> Microsoft-Windows-AppXDeploymentServer/Operational
Look for the latest log with an `error` level

**Reply by herrwinfried** [reply]
> Win+R run `eventvwr.msc` to open the event viewer In the left column find Application and Service Logs -> Microsoft -> Windows -> AppXDeployment-Server -> Microsoft-Windows-AppXDeploymentServer/Operational Look for the latest log with an `error` level

`AppX deployment failed for package MicrosoftCorporationII.WindowsSubsystemForAndroid_2207.40000.8.0_x64__8wekyb3d8bbwe error 0x80073CF6. The specific error text for this failure is: error 0x80070002: while preparing the request, the system was unable to register the windows.comInterface extension due to the following error: The system cannot find the file specified.`

**Reply by Howard20181** [reply]
There may be missing files, regenerate the WSA and try again.
